components/engine/libcontainerd/remote_unix.go

```diff
@@ -128,6 +136,34 @@ func New(stateDir string, options ...RemoteOption) (_ Remote, err error) {
 	return r, nil
 }
 
+func (r *remote) makeUpgradeProof() {
+	f, err := os.Open(runcStateDir)
+	if err != nil {
+		logrus.Warnf("libcontainerd: makeUpgradeProof could not find %s", runcStateDir)
+		return
+	}
+	fis, err := f.Readdir(0)
+	if err != nil {
+		logrus.Warnf("libcontainerd: makeUpgradeProof could not read directory entries in %s", runcStateDir)
+		f.Close()
+		return
+	}
+	ids := make([]string, len(fis))
+	for i, fi := range fis {
+		ids[i] = fi.Name()
+	}
+	f.Close()
+	for _, id := range ids {
```

Comment:
I don't think we should upgrade all the entries, we should only upgrade the one which ids are found in `/run/docker/libcontainerd/containerd`

Nothing prevent the user to having different version of `containerd`/`runc` on the machine and we don't want to break the containers started with those.